howto/android/odm_vhal.md

```diff
@@ -0,0 +1,133 @@
+*since 1.22.0*
+
+[note type="information" status="Note"]Replacing the Anbox VHAL is only supported on AAOS images.
+The Anbox Cloud dashboard does not support custom VHAL implementations.[/note]
+
+This document will guide through the process of replacing the Anbox Cloud VHAL
+implementation with your own implementation placed in the
+[ODM partition](https://source.android.com/docs/core/architecture/partitions/odm-partitions)
+using an [addon](https://discourse.ubuntu.com/t/38727).
+
+## Prerequisites
+
+Your custom VHAL implementation must provide an
+[HIDL VHAL interface](https://source.android.com/docs/automotive/vhal/hidl-vhal-interface)
+and be built against VNDK 29.
+
+If your VHAL implementation requires additional libraries, they must also be
+bundled with the addon and copied in the ODM partition.
+
+## Create the addon
+
+This addon will be tasked with the following:
+
+1. Copying the VHAL binary to `/odm/bin/hw`.
+1. Copying an
+[init.rc file](https://android.googlesource.com/platform/system/core/+/master/init/README.md)
+for your VHAL service to `/odm/etc/init`.
+1. Disabling the Anbox Cloud VHAL by setting the `ro.anbox.automotive.vhal`
+property to `odm`. This will be done by writing that system property to the
+`/odm/etc/build.prop` file.
+
+In the following example, we will name our addon `custom-vhal`.
+To override the VHAL implementation, we will use a
+[pre-start hook](https://discourse.ubuntu.com/t/28555) which is executed
+before Android gets started.
+
+The directory layout for the addon is strict and must be:
+
+```
+custom-vhal/
+├── hooks
+│   └── pre-start
+├── manifest.yaml
+├── vhal
+└── vhal.rc
+```
+
+As mentioned in the prerequisites, any additional libraries must also be bundled with the addon.
+
+`vhal` is your custom VHAL implementation.
+
+`vhal.rc` is an [init.rc file](https://android.googlesource.com/platform/system/core/+/main/init/README.md) for your VHAL service, such as:
+
+```
```

Comment:
Maybe it's worth adding a condition here to start the custom-vhal only when `ro.anbox.automotive.vhal` is set to odm. Otherwise, the default vhal implementation we provided may launch alongside custom-vhal.

Reply:
Hum, that shouldn't happen as `ro.anbox.automotive.vhal` would always be set to `odm` by `/odm/etc/build.prop`

I agree that it's a bit safer, though, I will add it.